docs/badges/details.rst

```diff
@@ -0,0 +1,168 @@
+Implementation Details
+======================
+
+Badges feature is implemented as a separate application within the Credentials Open edX service:
+
+.. code-block::
+
+        credentials/apps/badges/
+
+Events (Event Bus)
+------------------
+
+        Badges feature always relies on events (public signals) with user data.
+
+New events
+~~~~~~~~~~
+
+Badges feature extends the set of already `published events`_ with its own set of new public events:
+
+- `BADGE_AWARDED` (Credentials)
+- `BADGE_REVOKED` (Credentials)
+
+In addition to already present `public events`_ Badges also relies on (but event types set may be extended):
+
+- `COURSE_GRADE_NOW_PASSED` (LMS)
+- `COURSE_GRADE_NOW_FAILED` (LMS)
+
+
+Data models
+-----------
+
+Badges application maintains a set of data models.
+
+BadgeTemplate
+~~~~~~~~~~~~~
+
+**BadgeTemplate** data model extends `AbstractCredential`. It represents *native* badge templates.
+(additionally see *distribution badge templates*).
+
+.. code-block:: python
+
+        BadgeTemplate(AbstractCredential):
+        """
+        Describes badge credential type.
+        """
+        - uuid
+        - type: default="openedx" (native type)
+        - status: "inactive" | "active" | "archived"
+        - name
+        - description
+        - icon
+
+Requirement
+~~~~~~~~~~~
+
+**Requirement** describes an association between a BadgeTemplate and an Event.
```

Comment:
`Event` can be an overloaded term -- is the `Event` referenced here separate from the event bus and the Open edX events?